builders/Dockerfile.mingw64

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright 2022 The Geany contributors
+# License: GPLv2
+#
+# Docker image for Geany and Geany-Plugins cross-build to Windows
+# The image contains a self-compiled Pacman to install mingw-w64
+# packages and all other dependencies necessary to build the code
+# and create a ready-use installer.
+# For more details, see build_mingw64_geany.sh where this image is used.
+#
+# Intermediate container for building pacman
+FROM debian:bullseye as build-pacman
+
+ENV PACMAN_VERSION=6.0.1
+ENV PACMAN_SHA256="0db61456e56aa49e260e891c0b025be210319e62b15521f29d3e93b00d3bf731"
+ENV MSYS2_KEYRING_PKG="msys2-keyring-1~20220623-1-any.pkg.tar.zst"
+ENV MSYS2_KEYRING_PKG_SHA256="3508c7fca2f8b9722139666459eb8716f2413fd6daaf997abf0df41d7f285dc9"
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/pacman/lib/x86_64-linux-gnu
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get install --no-install-recommends --assume-yes \
+    build-essential meson wget xz-utils zstd gnupg2 file zstd ca-certificates \
+    pkg-config m4 libarchive-dev libssl-dev libcurl4-gnutls-dev libgpgme-dev \
+    python3-setuptools
+
+# compile Pacman
+RUN set -ex && \
+    wget --no-verbose https://sources.archlinux.org/other/pacman/pacman-${PACMAN_VERSION}.tar.xz && \
+    echo "${PACMAN_SHA256} *pacman-${PACMAN_VERSION}.tar.xz" | sha256sum --check --strict - && \
+    tar xf pacman-${PACMAN_VERSION}.tar.xz && \
+    cd /pacman-${PACMAN_VERSION} && \
+    meson \
+        --prefix /usr/local/pacman \
```

Comment:
I prefer to install applications into their own prefix if possible. But /usr/local should work as well.
Since the current setup works this way, I'd leave as is.

What do you mean by "potential binary-relocation issues in the future"?

Reply:
I don't understand. If you prefer the app-exclusive prefix, then why do you copy it to /usr/local/bin, afterwards?

It appears that pacman is run from /usr/local/bin/pacman, afterwards, yet its prefix is /usr/local/pacman. This can cause trouble if the pacman binary encodes the prefix (remember that Geany isn't relocatable by default either).

Really, I would prefer to define one prefix and then also run the program from there. Anything else is asking for trouble and is just adding confusion (I am confused). Then you don't need `ln -s` and `LD_LIBRARY_PATH` workarounds that I find in this file.

Reply:
Whateveryou wish.
Pacman is now installed into /usr/local but setting `LD_LIBRARY_PATH` is still necessary or alternatively running `ldconfig` to manually update the ldcache.
This is because after installing Pacman with Meson, the ldcache is not updated automatically. So we still need to take care to update the ldcache.

Reply:
Can you run ldconfig in the container?

Reply:
Yes, tried it in the first place. We would need to run it in the build container as well as in the final container (as we copy only the files in /usr/local from the one to the other but no state and cache).
I don't mind whether we set `LD_LIBRARY_PATH` as it is or run `ldconfig`.

Reply:
My feeling: better update the cache properly than messing with LD_LIBRARY_PATH but I don't feel strong either.

Reply:
Ok, done.